phpBB/phpbb/notification/type/mention.php

```diff
@@ -0,0 +1,397 @@
+	{
+		$this->user_loader = $user_loader;
+	}
+
+	/**
+	* Is available
+	*/
+	public function is_available()
+	{
+		return $this->config['allow_topic_notify'];
+	}
+
+	/**
+	* Get the id of the item
+	*
+	* @param array $post The data from the post
+	* @return int The post id
+	*/
+	static public function get_item_id($post)
+	{
+		return (int) $post['post_id'];
+	}
+
+	/**
+	* Get the id of the parent
+	*
+	* @param array $post The data from the post
+	* @return int The topic id
+	*/
+	static public function get_item_parent_id($post)
+	{
+		return (int) $post['topic_id'];
+	}
+
+	/**
+	* Get the user's avatar
+	*/
+	public function get_avatar()
+	{
+		return $this->user_loader->get_avatar($this->get_data('poster_id'), false, true);
+	}
+
+	/**
+	* Retrieves an ArrayIterator over the configuration values.
+	*
+	* @return \ArrayIterator An iterator over all config data
+	*/
+	public function getIterator()
+	{
+		return new \ArrayIterator($this->config);
+	}
+
+		public function find_users_for_notification($db, $users_to_notify = array())
+	{
+		$sql_ary = "SELECT users.user_id, users.username_clean FROM ". USERS_TABLE . " as users, " . USER_NOTIFICATIONS_TABLE . " as notif WHERE " . $db->sql_in_set('username_clean', $users_to_notify) . " AND users.user_id = notif.user_id AND notif.item_type = 'notification.type.mention' " ;
+		$result = $db->sql_query($sql_ary);
+		$user_list = array();
+		while ($row = $db->sql_fetchrow($result))
+		{
+
+			$user_list[$row["username_clean"]] = $row["user_id"];
+		}
+		$db->sql_freeresult($result);
+		return $user_list;
+	}
+
+	public function get_notification_type_and_method($db, $user_list, $notif_manager)
+	{
+		$integer_ids = array_map('intval', $user_list);
+		$sql_query = "SELECT user_id, item_type, method FROM " . USER_NOTIFICATIONS_TABLE . " where " . $db->sql_in_set("user_id", $integer_ids) . " AND item_type = 'notification.type.mention'";
+		$result = $db->sql_query($sql_query);
+		$notif_list = array();
+		while ($row = $db->sql_fetchrow($result))
+		{
+			$temp_notif_list = array();
+			$temp_notif_list["user_id"] = $row["user_id"];
+			$temp_notif_list["notif_type"] = $notif_manager->get_item_type_class($row["item_type"]);
+			$temp_notif_list["notif_method"] = $notif_manager->get_method_class($row["method"]);
+			array_push($notif_list, $temp_notif_list);
+		}
+		$db->sql_freeresult($result);
+		return $notif_list;
+	}
+
+
+	/**
+	* Get the url to this item
+	*
+	* @return string URL
+	*/
+	public function get_url()
+	{
+		return append_sid($this->phpbb_root_path . 'viewtopic.' . $this->php_ext, "p={$this->item_id}#p{$this->item_id}");
+	}
+
+	/**
+	* {inheritDoc}
+	*/
+	public function get_redirect_url()
+	{
+		return append_sid(phpbb_root_path . 'viewtopic.' . $this->php_ext, "t={$this->item_parent_id}&amp;view=unread#unread");
+	}
+
+	/**
+	* Get email template
+	*
+	* @return string|bool
+	*/
+	public function get_email_template()
+	{
+		return 'topic_notify';
```

Comment:
Create a new email template to send, as it is something different as notifying for the topic :)